CodePush.js

```diff
@@ -41,7 +40,7 @@ async function checkForUpdate(deploymentKey = null) {
    * in any updates for current app store version, regardless of hash.
    */
   let queryPackage;
-  if (localPackage && localPackage.appVersion && semver.compare(localPackage.appVersion, config.appVersion) === 0) {
+  if (localPackage && localPackage.appVersion && localPackage.appVersion === config.appVersion) {
```

Comment:
As I think about this more, since we already clear updates when a new binary is installed in release, the only time that this last expression would matter is after a binary update in debug mode. In those scenarios, it's likely that the dev didn't bump the version anyways, and therefore, this version check would only prevent using an update in debug mode, where the dev actually bumped the version. If we care about this scenario, I would argue that it might be a cleaner solution to change our logic to clear updates in debug mode if the version has changed.